exploration-module/src/main/java/io/edpn/backend/exploration/application/dto/v1/SystemDTO.java

```diff
@@ -0,0 +1,22 @@
+package io.edpn.backend.exploration.application.dto.v1;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.extern.jackson.Jacksonized;
+
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Jacksonized
+public class SystemDTO {
```

Comment:
Personal preferences (ie. my dislike of annotations and other "magic" that obscures things) aside: why setters _and_ builder, and all-args _and_ no-args? IMHO, it's usually mutable -> setters, n-oargs; immutable -> no setters, all-args, builder (if there's a large number of fields).

Reply:
Also, `record`s are wonderful and should, unless there's a very specific reason not to, be embraced! :hugs:

Reply:
Setters for any @Valid mapping of GET Params to an object. You do not use @RequestBody for this, and this requires set methods.

@Builder allows us to write extremely clean mappers.